[Doc] Improve docs on stack trace capturing (#1205)
elastic/apm-agent-dotnet: docs/troubleshooting.asciidoc

and then running the application in a context where the environment variable will be visible. In setting this value,
an `ElasticApmAgentStartupHook.log` file is written to in the directory containing the startup hook assembly, in addition to
writing to standard output.

[float]
[[agent-overhead]]
=== The agent causes too much overhead

> A good place to start is our <<config-all-options-summary>>. There are multiple settings there with the `Performance` keyword which help you to tweak the agent for your needs.

Reviewer: ```suggestion
A good place to start is <<config-all-options-summary>>. There are multiple settings with the `Performance` keyword which can help to tweak the agent for your needs.
```

The most expensive operation in the agent is typically stack trace capturing. The agent, by default only captures stack traces for spans with a duration of 5ms or more, and with a limit of 50 stack frames.

> 

Reviewer 2: ```suggestion
```

The sentences before and after this line rely on each other. I'd remove this break. You also might consider bulleting out your suggestions and leading with the outcome instead of the configuration variable. In other words, something like this:

> **The agent causes too much overhead**
> A good place to start is [config-all-options-summary]. There are multiple settings there with the `Performance` keyword which can help you tweak the agent for your needs.
> 
> The most expensive operation in the agent is typically stack trace capturing. The agent, by default, only captures stack traces for spans with a duration of 5ms or more, and with a limit of 50 stack frames.
> If this is too much in your environment, consider disabling stack trace capturing either partially or entirely:
> 
> * To disable stack trace capturing for spans, but continue to capture stack traces for errors, set the [config-span-frames-min-duration] to `0` and leave the [config-stack-trace-limit] on its default.
> * To disable stack trace capturing entirely–which in most applications reduces the agent overhead dramatically–set [config-stack-trace-limit] to `0`.

> If this is too much in your environment, the <<config-span-frames-min-duration>> and the <<config-stack-trace-limit>> are two configuration worth looking at.

Reviewer: ```suggestion
If this is too much in your environment, the <<config-span-frames-min-duration>> and the <<config-stack-trace-limit>> are two configuration settings worth looking at.
```

> By setting <<config-stack-trace-limit>> to `0` you can disable stack trace capturing completely which in most applications reduces the agent overhead dramatically.

Reviewer: ```suggestion
By setting <<config-stack-trace-limit>> to `0`, stack trace capturing is completely disabled, which reduces the agent overhead dramatically for most applications.
```

> If you would like to capture stack traces for errors but skip stack trace capturing for spans, you can set the <<config-span-frames-min-duration>> to `0` and leave the <<config-stack-trace-limit>> on default.

Reviewer: ```suggestion
To **capture stack traces only for errors** and not for spans, set the <<config-span-frames-min-duration>> to `0` and leave the <<config-stack-trace-limit>> as the default.
```
